src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -207,9 +214,20 @@ internal static class PipelineOps
             bool redirectedInformation = false;
             if (redirections != null)
             {
+                foreach (var redirection in redirections) {
+                    if (redirection is MergingRedirection)
+                    {
+                        redirection.Bind(pipe, commandProcessor, context);
+                    }
+                }
```

Comment:
The double enumeration here was a quick band-aid for `native 2>&1 > file.txt`. In that scenario we want to revert to the current string reading behavior so I needed to process merging redirections before I tried to create a specialized byte pipe for a file stream.

I need to find implement a better pattern here.